modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/ElixirClientCodegen.java

```diff
@@ -590,9 +590,26 @@ private void buildTypespec(CodegenParameter param, StringBuilder sb) {
                 buildTypespec(param.items, sb);
                 sb.append("}");
             } else if (param.isPrimitiveType) {
-                // <type>.t
-                sb.append(param.dataType);
-                sb.append(".t");
+                // <type>() OR <type>.t
+
+                // Primitive types in Elixir
+                // https://hexdocs.pm/elixir/1.5.2/typespecs.html#types-and-their-syntax
+                //
+                // NOTE: List, Tuple and Map are declared as primitive in a variable `languageSpecificPrimitives`.
+                HashMap map = new HashMap<String, String>();
```

Comment:
Not a java expert, but this mapping could probably be shared across buildTypespec calls.